src/cli/upgrade.rs

```diff
@@ -30,18 +31,51 @@ pub struct UpgradeCommandArguments {
 impl UpgradeCommandArguments {
 	/// Get the version string to download based on the user preference
 	async fn version(&self) -> Result<Cow<'_, str>, Error> {
-		if self.nightly {
-			Ok(Cow::Borrowed("nightly"))
-		} else if self.beta {
-			fetch("beta").await
-		} else if let Some(version) = self.version.as_ref() {
-			Ok(Cow::Borrowed(version))
+		let nightly = "nightly";
```

Comment:
These could be consts outside the function. Unless this is deliberate for the Cow?

Reply:
Yes @phughk, I did consider that and that's what I wanted to do at first but having these as constants but not being able to then use those constants to replace the strings in `conflicts_with = "nightly"` and `conflicts_with = "beta"` was annoying me. I know it's petty! 🤣 Also, the constants would only be used in this method.

Reply:
Ah so I was wrong! Thanks for explanation, happy with the approach :)